flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/match/RowtimeProcessFunction.java

```diff
@@ -44,7 +44,7 @@ public RowtimeProcessFunction(int rowtimeIdx, TypeInformation<BaseRow> returnTyp
 
 	@Override
 	public void processElement(BaseRow value, Context ctx, Collector<BaseRow> out) throws Exception {
-		long timestamp = value.getLong(rowtimeIdx);
+		long timestamp = value.getTimestamp(rowtimeIdx, 3).getMillisecond();
```

Comment:
Can Timestamp with 9 be a time attribute?

Reply:
I'm afraid it can not.

Reply:
+1 we should pass in the precision too.

Reply:
Looks like it is very easy to pass the precision.
So I would suggest like Jark, just pass it to avoid more hardcode.

Reply:
OK, it's make sense to pass the precision